### PR TITLE
Include a mermaid architecture diagram

### PR DESCRIPTION
**Pull request by stevemar**
This PR includes an architecture diagram created with mermaid.js.
Additionally, this PR cleans up some of the top level README.

Related: #35

Signed-off-by: Steve Martinelli <steve.martinelli@shopify.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,36 +2,58 @@
...
 
-This sample app was built as a reference for how Shopify Developer tools can be used together to create a fully functional application.
+This sample app was built as a reference to show how Shopify Developer tools can be used together to create a [fully functional application](https://shopify.dev/apps/). The example we chose to showcase is an application that facilitates review creation for products in a Shopify store.
 
-The example we chose to showcase is an app which facilitates review creation for products in a Shopify store.
+The application is written in [Node.js](https://nodejs.org/) and is comprised of the following elements:
 
-It makes use of Metafields as a way to store the reviews, which means it requires no database.
+- **[Embedded Node.js application](https://shopify.dev/apps/getting-started/create)**: Which displays an admin UI for managing (publishing, unpublishing, deleting) product reviews.
+- **[Theme app extension](https://shopify.dev/apps/online-store/theme-app-extensions)**: Which enables a merchant to include review information, per product, to their online store.
```
Comment (meganmajewski) [inline README.md:10]: Maybe change include to display?

File: README.md [hunk 2]
```diff
@@ -2,36 +2,58 @@
...
+        meta[Metafield API]
+        products[Product API]
+        oauth[OAuth API]
+    end
+
+     classDef green fill:#9f6,stroke:#333,stroke-width:2px;
+     classDef orange fill:#f96,stroke:#333,stroke-width:2px;
+     class te,ratings green
+     class feedback,pc orange
+```
```
Comment (meganmajewski) [inline README.md:41]: 😍

File: README.md [hunk 3]
```diff
@@ -2,36 +2,58 @@
...
-We've created this app to inspire the next generation of apps developed for Shopify merchants. When learning from this code, keep in mind:
+When using this code keep in mind:
 
-1. This is a sample app, not a production ready app.
-2. Metafields are going to be a much bigger part of the platform and will support building production apps with a lot of data. For now, we recommend metafields for simpler data for apps.
+1. This is a sample application. It is not production-ready.
+2. Metafields are going to be a bigger part of the platform and will support building applications with more complex data structures.
 
-## App developer requirements
+## Requirements
```
[the remark this answers is not in the document]
Comment (stevemar) [reply README.md:50]: Eh, I think we'll end up collapsing those tbh. I'll start working on that next.

**Comment by stevemar** [comment]
@meganmajewski pushed a new revision.

**Comment by meganmajewski** [comment]
looks good!

**Pull request merged**